ecs-agent/ecs/model/api/docs-2.json

```diff
@@ -0,0 +1,4197 @@
+    "DiscoverPollEndpoint": "<note> <p>This action is only used by the Amazon ECS agent, and it is not intended for use outside of the agent.</p> </note> <p>Returns an endpoint for the Amazon ECS agent to poll for updates.</p>",
+    "ExecuteCommand": "<p>Runs a command remotely on a container within a task.</p> <p>If you use a condition key in your IAM policy to refine the conditions for the policy statement, for example limit the actions to a specific cluster, you receive an <code>AccessDeniedException</code> when there is a mismatch between the condition key value and the corresponding parameter value.</p> <p>For information about required permissions and considerations, see <a href=\"https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-exec.html\">Using Amazon ECS Exec for debugging</a> in the <i>Amazon ECS Developer Guide</i>. </p>",
+    "GetDeployment": null,
+    "GetTaskProtection": "<p>Retrieves the protection status of tasks in an Amazon ECS service.</p>",
+    "ListAccountSettings": "<p>Lists the account settings for a specified principal.</p>",
+    "ListAttributes": "<p>Lists the attributes for Amazon ECS resources within a specified target type and cluster. When you specify a target type and cluster, <code>ListAttributes</code> returns a list of attribute objects, one for each attribute on each resource. You can filter the list of results to a single attribute name to only return results that have that name. You can also filter the results by attribute name and value. You can do this, for example, to see which container instances in a cluster are running a Linux AMI (<code>ecs.os-type=linux</code>). </p>",
+    "ListClusters": "<p>Returns a list of existing clusters.</p>",
+    "ListContainerInstances": "<p>Returns a list of container instances in a specified cluster. You can filter the results of a <code>ListContainerInstances</code> operation with cluster query language statements inside the <code>filter</code> parameter. For more information, see <a href=\"https://docs.aws.amazon.com/AmazonECS/latest/developerguide/cluster-query-language.html\">Cluster Query Language</a> in the <i>Amazon Elastic Container Service Developer Guide</i>.</p>",
+    "ListDeployments": null,
+    "ListServices": "<p>Returns a list of services. You can filter the results by cluster, launch type, and scheduling strategy.</p>",
+    "ListServicesByNamespace": "<p>This operation lists all of the services that are associated with a Cloud Map namespace. This list might include services in different clusters. In contrast, <code>ListServices</code> can only list services in one cluster at a time. If you need to filter the list of services in a single cluster by various parameters, use <code>ListServices</code>. For more information, see <a href=\"https://docs.aws.amazon.com/AmazonECS/latest/developerguide/service-connect.html\">Service Connect</a> in the <i>Amazon Elastic Container Service Developer Guide</i>.</p>",
```

Comment:
It seems like this `ListServicesByNamespace` is not present in `api-2.json`. Was it removed (intentionally/unintentionally)? If it was removed intentionally, we should remove it from here as well. If it was removed unintentionally, we should make sure it is present in `api-2.json`.

Reply:
ListServicesByNamespace had the same issue as the other shapes mentioned [here](https://github.com/aws/amazon-ecs-agent/pull/3974#discussion_r1370749152). ListServicesByNamespace was referencing a shape which was internal only. Instead of removing the just the error being referenced, the whole operation was removed. I will discuss with you offline on the correct behavior as mentioned in this [comment](https://github.com/aws/amazon-ecs-agent/pull/3974#discussion_r1370759552) as well.